### PR TITLE
[FlowCleanup] Split ImageProps into DeprecatedPropTypes

### PR DESCRIPTION
**Pull request by sopranolinist**
Related to #21342

- Split ImageProps.js: moved propType declarations to DeprecatedImageProps.js
- Renamed ImageProps references to DeprecatedImageProps in Image.ios.js

Test Plan:
----------
Ran the following scripts with no errors:
- `npm run prettier`
- `npm run flow-check-ios`
- `npm run flow-check-android`

Release Notes:
--------------

[GENERAL] [ENHANCEMENT] [ImageProps.js] - rm prop-types
[GENERAL] [ENHANCEMENT] [DeprecatedImageProps.js] - created file & added prop-types
[GENERAL] [ENHANCEMENT] [Image.ios.js] - renamed ImageProps to DeprecatedImageProps

This is my first PR, please let me know if I've done anything incorrectly - thanks!

### CODE REVIEW

File: Libraries/DeprecatedPropTypes/DeprecatedImageProps.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+const DeprecatedImageSourcePropType = require('DeprecatedImageSourcePropType');
+const DeprecatedImageStylePropTypes = require('DeprecatedImageStylePropTypes');
+const DeprecatedStyleSheetPropType = require('DeprecatedStyleSheetPropType');
+const PropTypes = require('prop-types');
+
+module.exports = {
+  /**
+   * See https://facebook.github.io/react-native/docs/image.html#style
+   */
+  style: DeprecatedStyleSheetPropType(DeprecatedImageStylePropTypes),
```
Comment (RSNara) [inline Libraries/DeprecatedPropTypes/DeprecatedImageProps.js]: In general, we should have a singular source of truth for the documentation, which should be the flow types. So, could you move these comments to the `ImageProps` flow type?

File: Libraries/DeprecatedPropTypes/DeprecatedImageProps.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+'use strict';
```
[the remark this answers is not in the document]
Comment (sopranolinist) [reply Libraries/DeprecatedPropTypes/DeprecatedImageProps.js]: Done - will include in next push.

File: Libraries/DeprecatedPropTypes/DeprecatedImageProps.js (hunk 1 above)
Comment (sopranolinist) [reply Libraries/DeprecatedPropTypes/DeprecatedImageProps.js]: While moving comments over, I noticed a few discrepancies between Flow properties and propType properties. The following are declared as propTypes (in DeprecatedImageProps, soon to be renamed DeprecatedImagePropType) but not in Flow (in ImageProps):
* accessible: PropTypes.bool
* accessibilityLabel: PropTypes.node
* resizeMode: PropTypes.oneOf([
    'cover',
    'contain',
    'stretch',
    'repeat',
    'center',
  ])
* testID: PropTypes.string
* onLayout: PropTypes.func

Are these not needed in the Flow definitions, or are they missing from the Flow definitions?

Comment (RSNara) [reply Libraries/DeprecatedPropTypes/DeprecatedImageProps.js]: Oh, interesting. We should move these properties to the flow definitions.

**Pull request closed without merging**